activemodel/lib/active_model/attribute_methods.rb

```diff
@@ -72,6 +72,8 @@ module AttributeMethods
     included do
       class_attribute :attribute_aliases, instance_writer: false, default: {}
       class_attribute :attribute_method_patterns, instance_writer: false, default: [ ClassMethods::AttributeMethodPattern.new ]
+
+      include BeforeTypeCast
```

Comment:
This should not be in `AttributeMethods`. This module doesn't do any type cast. It makes sense to be in `ActiveModel::Model` when `Attributes` is included by default or in `Attributes` since that is the module that knows about type cast.

Reply:
@rafaelfranca I think you are agreeing with what I said in https://github.com/rails/rails/pull/44534#issuecomment-1049085235, yes?  I've made that change now.

Also, I've kept an `ActiveRecord::AttributeMethods::BeforeTypeCast` placeholder module like I suggested in https://github.com/rails/rails/pull/44534#issuecomment-1049127985, but I can remove it, if you prefer.